src/Microsoft.DotNet.Wpf/src/PresentationFramework/System/Windows/Documents/TextRangeSerialization.cs

```diff
@@ -572,9 +572,7 @@ private static void WriteStartXamlElement(ITextRange range, ITextPointer textRea
             if (customTextElement)
             {
                 // If the element is not from PresentationFramework, we'll need to serialize a namespace
-                // Will module name always have a '.'?  If so, can remove conditional in assembly assignment below
-                int index = elementTypeStandardized.Module.Name.LastIndexOf('.');
-                string assembly = (index == -1 ? elementTypeStandardized.Module.Name : elementTypeStandardized.Module.Name.Substring(0, index));
+                string assembly = elementTypeStandardized.Assembly.GetName().Name;
```

Comment:
Another case where the two methods of getting the `assembly` string (probably) return different results.  This one matters a lot too - the serialized result gets de-serialized by some other process, and presumably the old code uses the substring method to make that de-serialization work correctly in all cases.

Again, use the old code, falling back to the new code in the single-file case.  And find out what breaks, so we can warn people.

Reply:
In my rudimentary testing the assembly and the trimmed module matched in name
```C#
Assembly? asm = Assembly.GetAssembly(typeof(UnconditionalSuppressMessageAttribute));
foreach (Module module in asm!.Modules)
    Console.WriteLine(module.Name.Substring(0, module.Name.LastIndexOf(".")));
Console.WriteLine(asm.GetName().Name);
#System.Private.CoreLib
#System.Private.CoreLib
```
But yes, I'm not 100% sure that is always the case. I put a fallback scenario that uses the assebly name only in the case in which Module.Name is <Unknown>